CONTRIBUTING.rst

```diff
@@ -254,6 +255,15 @@ Guidelines
           # function not for directed graphs *or* for multigraphs
           pass
 
+* Adding new default and extra dependencies should be done only with careful
+  consideration.  New default dependencies should be widely-used in the community
+  and have high potential for wide-spread use in NetworkX.  They should also be
+  pure Python and easy to install.  Extra dependencies should also be easy to
```

Comment:
I wasn't sure about this.  I was thinking we might be reluctant to add non-pure Python packages (besides numpy, scipy, etc.), since it would be even more unlikely to work on PyPy or other non-CPython platforms.

That said, I am not committed to this one way or the other.  (I was mainly trying to be a bit conservative, since this is already a big break from the last 15 years or so of networkx.)  So if Dan, Mridul, and you are happy with taking out ``pure Python and`` that is fine with me.